src/Core/src/Handlers/Entry/EntryHandler.Android.cs

```diff
@@ -209,6 +209,10 @@ void ITextWatcher.BeforeTextChanged(Java.Lang.ICharSequence? s, int start, int c
 
 			void ITextWatcher.OnTextChanged(Java.Lang.ICharSequence? s, int start, int before, int count)
 			{
+				// we are replacing 0 characters with 0 characters, so skip
+				if (before == 0 && count == 0)
+					return;
```

Comment:
Setting InputType triggers a non-text changing event. This might result in a case where an event comes and the text is not yet set. Like on this init. Instead of assuming the text is changing, make sure that it is before proceeding.